apps/civicsignalblog/src/payload/components/actions/index.tsx

```diff
@@ -0,0 +1,98 @@
+import { useAuth } from "payload/components/utilities";
+import React, { useEffect, useState } from "react";
+
+import applications, { RESEARCH } from "../../lib/data/common/applications";
+
+function BeforeDashboard() {
+  const { user } = useAuth();
+
+  const [selectedApp, setSelectedApp] = useState(
+    user.currentApp || user.defaultApp || RESEARCH,
+  );
+
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (
+      selectedApp &&
+      selectedApp !== "select" &&
+      selectedApp !== String(user.currentApp)
+    ) {
+      setLoading(true);
+      fetch(`/api/users/apps/current`, {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ selectedApp }),
+      })
+        .then((response) => {
+          setLoading(false);
+          if (response.ok) {
+            return response.json();
+          } else {
+            throw new Error("Network response was not ok");
+          }
+        })
+        .then((data) => {
+          window.location.replace(`/admin?app=${encodeURI(data.currentApp)}`);
+        })
+        .catch((error) => {
+          setLoading(false);
+          throw error;
+        });
```

Comment:
Minor but shouldn't we be using `async`? If we **_have to_** use promise, shouldn't there just be a single `finally` that sets loading to false?